libtsuba/src/PartitionTopologyMetadata.cpp

```diff
@@ -0,0 +1,75 @@
+#include "PartitionTopologyMetadata.h"
+
+#include <memory>
+#include <string>
+
+#include <boost/format.hpp>
+
+#include "katana/Logging.h"
+
+namespace tsuba {
+
+PartitionTopologyMetadataEntry*
+PartitionTopologyMetadata::GetEntry(uint32_t index) {
+  KATANA_LOG_ASSERT(index < num_entries_);
+  return &(entries_.at(index));
+}
+
+PartitionTopologyMetadataEntry*
+PartitionTopologyMetadata::Append(PartitionTopologyMetadataEntry entry) {
+  KATANA_LOG_ASSERT(num_entries_ < max_num_topologies);
+  entries_[num_entries_] = std::move(entry);
+
+  num_entries_++;
+  return &(entries_.at(num_entries_ - 1));
+}
+
+katana::Result<void>
+PartitionTopologyMetadata::Validate() const {
+  // basic validation, a CSR topology must be present
+  if (entries_.empty()) {
+    return KATANA_ERROR(
+        ErrorCode::InvalidArgument, "no topology metadata entries present");
+  }
+  for (size_t i = 0; i < num_entries_; i++) {
+    auto res = ValidateEntry(entries_.at(i));
+    if (!res) {
+      return res;
+    }
+  }
+  return katana::ResultSuccess();
+}
+
+std::string
+PartitionTopologyMetadata::EntryToString(
+    const PartitionTopologyMetadataEntry& entry) const {
+  boost::format fmt = boost::format(
+                          "transpose_%1%_node_sort_%2%_edge_sort_%3%_num_nodes_"
+                          "%4%_num_edges_%5%") %
+                      entry.transpose_state_ % entry.node_sort_state_ %
+                      entry.edge_sort_state_ % entry.num_nodes_ %
+                      entry.num_edges_;
+  return fmt.str();
+}
+
+katana::Result<void>
+PartitionTopologyMetadata::ValidateEntry(
+    const PartitionTopologyMetadataEntry& entry) const {
+  if (entry.path_.empty()) {
+    std::string name = EntryToString(entry);
+    return KATANA_ERROR(
+        ErrorCode::InvalidArgument, "no topology file path: topology: {} ",
+        name);
+  }
+  if (entry.path_.find('/') != std::string::npos) {
```

Comment:
It looks like this will return an error if '/' _is_ found?

Reply:
Agree it's a bad error, not a bad check.